tests/lib/rules/require-expect.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * @fileoverview Require the use of `expect` when using `assert` inside of a
+ * block or when passing `assert` to a function.
+ * @author Mitch Lloyd
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var rule = require("../../../lib/rules/require-expect"),
+    RuleTester = require("eslint").RuleTester;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+var ruleTester = new RuleTester(),
+    returnAndIndent = "\n        ",
+    ALWAYS_ERROR_MESSAGE = "Should use `assert.expect` when using assertions outside of the top-level test callback",
+    EXCEPT_SIMPLE_ERROR_MESSAGE = "Test is missing `assert.expect() call`";
+
+ruleTester.run("require-expect", rule, {
+
+    valid: [
+        // default, calling expect is valid
+        {
+            code: "test('name', function(assert) { assert.expect(0) });",
+            options: []
+        },
+
+        // CallExpression with no object doesn't error
+        {
+            code: "test('name', function(assert) { assert.expect(0); other() });",
+            options: []
+        },
+
+        // assert at top of test context is ok
+        {
+            code: "test('name', function(assert) { assert.ok(true) });",
+            options: ["except-simple"]
+        },
+
+        // assert with expect at the top of test context is ok
+        {
+            code: "test('name', function(assert) { assert.expect(0); if (false) { assert.ok(false) } });",
+            options: ["except-simple"]
+        },
+
+        // nested modules
+        {
+            code: [
+                "module('a', function() {",
+                "    test('a - test', function(assert) {",
+                "        assert.ok(true, 'no expect needed');",
+                "    });",
+                "",
+                "    module('b', function() {",
+                "        test('b - test', function(assert) {",
+                "            assert.ok(true, 'still no expect needed');",
+                "        });",
+                "    });",
+                "});"
+            ].join(returnAndIndent),
+            options: ["except-simple"]
+        }
+
+    ],
```

Comment:
Could you add some test cases for where the assert context variable is named something else? And I mentioned global `expect` as well-- that needs rigorous coverage too.